_posts/2023-01-18-v0.25-incremental-compilation-and-deno.md

```diff
@@ -0,0 +1,279 @@
+---
+author: Louis Pilfold
+title: Incremental compilation and hello Deno!
+subtitle: Gleam v0.26 released
+tags:
+  - Release
+---
+
+Gleam is a type safe and scalable language for the Erlang virtual machine and
+JavaScript runtimes. Today Gleam [v0.26.0][release] has been released, let's
+take a look at what's new.
+
+[release]: https://github.com/gleam-lang/gleam/releases/tag/v0.26.0
+
+
+## Incremental compilation
+
+A Gleam project is made of packages, typically a top level package and several
+dependency packages fetched by the package manager, and each package contains a
+collection of modules of Gleam code.
+
+In the very early days of Gleam when the compiler was run it would compile from
+scratch every module in every package in the project. This was highly wasteful,
+especially for the dependency packages which are would not have changed at all.
+To tackle this inefficiency when the Gleam build tool was created it was made to
+compile dependency packages only once and reuse the compiled code for every
+following build, resulting in only the top level package being recompiled.
+
+This has worked well for the last couple years, but now as more people are using
+Gleam it was time for an upgrade. Large projects such as those with a large
+amount of generated gRPC code were starting to take an irksome amount of time to
+compile. Gleam is all about fun and productivity, so this just won't do!
+
+There are numerous ways we want to improve the performance of the (already very
+nimble) Gleam compiler, but the majority of the time is spent in the Erlang
```

Comment:
Should there be a note somewhere around here or after this paragraph that we're looking at the Erlang target compilation as an example / target to optimize? I found it odd that the blog post introductory paragraph mentioned both Erlang and Javascript, but then Javascript was silently ignored

Reply:
We're looking to optimise it all. I've got unpushed changes that talks more about JavaScript also :)